javatests/arcs/showcase/ShowcaseEnvironment.kt

```diff
@@ -0,0 +1,180 @@
+@file:Suppress("EXPERIMENTAL_IS_NOT_ENABLED")
+
+package arcs.showcase
+
+import android.app.Application
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import androidx.test.core.app.ApplicationProvider
+import androidx.work.testing.WorkManagerTestInitHelper
+import arcs.android.storage.database.AndroidSqliteDatabaseManager
+import arcs.core.allocator.Allocator
+import arcs.core.allocator.Arc
+import arcs.core.data.Plan
+import arcs.core.host.AbstractArcHost
+import arcs.core.host.ArcHost
+import arcs.core.host.EntityHandleManager
+import arcs.core.host.ParticleRegistration
+import arcs.core.host.ParticleState
+import arcs.core.host.SchedulerProvider
+import arcs.core.storage.ActivationFactory
+import arcs.core.storage.api.DriverAndKeyConfigurator
+import arcs.jvm.host.ExplicitHostRegistry
+import arcs.jvm.host.JvmSchedulerProvider
+import arcs.jvm.util.JvmTime
+import arcs.sdk.Particle
+import arcs.sdk.android.storage.ServiceStoreFactory
+import arcs.sdk.android.storage.service.testutil.TestConnectionFactory
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+import kotlin.coroutines.EmptyCoroutineContext
+
+/**
+ * A JUnit rule setting up an Arcs environment for showcasing features.
+ *
+ * Usage example follows:
+ *
+ * ```
+ * @get:Rule val env = ShowcaseEnvironment(
+ *     ::SomeParticle.toRegistration(),
+ *     ::OtherParticle.toRegistration(),
+ * )
+ *
+ * @Test
+ * fun answerToEverythingIsCorrect() = runTest {
+ *   // Start an Arc.
+ *   val arc = env.startArc(YourGeneratedPlan)
+ *
+ *   // Do something with it.
+ *   env.getParticle<SomeParticle>(arc).query("what's the answer?")
+ *
+ *   // Assert result.
+ *   assertThat(env.getParticle<OtherParticle>(arc).state).isEqualTo(42)
+ * }
+ * ```
+ */
+@ExperimentalCoroutinesApi
+class ShowcaseEnvironment(
```

Comment:
Is there any convention about always naming `TestRule` classes with a `Rule` suffix?

Reply:
I don't think so, based on this: https://junit.org/junit4/javadoc/4.12/org/junit/rules/package-summary.html

You have: TemporaryFolder, ErrorCollector, Stopwatch etc.